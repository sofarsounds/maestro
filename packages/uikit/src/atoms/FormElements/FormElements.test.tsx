import React from 'react';
import { mountWithTheme } from '../../test';

import { Textfield, Textarea } from './index';

describe('FormElements', () => {
  describe('Textfield', () => {
    it('renders correctly', () => {
      expect(mountWithTheme(<Textfield />)).toMatchSnapshot();
    });

    it('renders correctly in error state', () => {
      expect(mountWithTheme(<Textfield hasError />)).toMatchSnapshot();
    });

    it('renders correctly with a value', () => {
      const wrapper = mountWithTheme(
        <Textfield disabled value="I am a textfield" />
      );
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(Textfield).prop('value')).toBe('I am a textfield');
    });

    it('renders correctly in disabled state', () => {
      const wrapper = mountWithTheme(<Textfield disabled value="Disabled" />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(Textfield).prop('disabled')).toBeTruthy();
    });
  });

  describe('Textarea', () => {
    it('renders correctly', () => {
      expect(mountWithTheme(<Textarea />)).toMatchSnapshot();
    });

    it('renders correctly in error state', () => {
      expect(mountWithTheme(<Textarea hasError />)).toMatchSnapshot();
    });

    it('renders correctly with a value', () => {
      const wrapper = mountWithTheme(
        <Textarea disabled value="I am a textarea" />
      );
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(Textarea).prop('value')).toBe('I am a textarea');
    });

    it('renders correctly in disabled state', () => {
      const wrapper = mountWithTheme(<Textarea disabled value="Disabled" />);
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(Textarea).prop('disabled')).toBeTruthy();
    });
  });
});
