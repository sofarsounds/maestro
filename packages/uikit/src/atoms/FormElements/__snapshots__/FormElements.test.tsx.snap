// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormElements Textarea renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  height: auto;
  min-height: 44px;
  resize: none;
  padding: 16px;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.textarea>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.textarea",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "textarea",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <textarea
        className="c0"
      />
    </StyledComponent>
  </styled.textarea>
</ThemeProvider>
`;

exports[`FormElements Textarea renders correctly in disabled state 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  height: auto;
  min-height: 44px;
  resize: none;
  padding: 16px;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.textarea
    disabled={true}
    value="Disabled"
  >
    <StyledComponent
      disabled={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.textarea",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "textarea",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      value="Disabled"
    >
      <textarea
        className="c0"
        disabled={true}
        value="Disabled"
      />
    </StyledComponent>
  </styled.textarea>
</ThemeProvider>
`;

exports[`FormElements Textarea renders correctly in error state 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  border-color: #EF4423;
  height: auto;
  min-height: 44px;
  resize: none;
  padding: 16px;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.textarea
    hasError={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.textarea",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "textarea",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      hasError={true}
    >
      <textarea
        className="c0"
      />
    </StyledComponent>
  </styled.textarea>
</ThemeProvider>
`;

exports[`FormElements Textarea renders correctly with a value 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  height: auto;
  min-height: 44px;
  resize: none;
  padding: 16px;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.textarea
    disabled={true}
    value="I am a textarea"
  >
    <StyledComponent
      disabled={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";

  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.textarea",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "textarea",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      value="I am a textarea"
    >
      <textarea
        className="c0"
        disabled={true}
        value="I am a textarea"
      />
    </StyledComponent>
  </styled.textarea>
</ThemeProvider>
`;

exports[`FormElements Textfield renders correctly 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.input>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <input
        className="c0"
      />
    </StyledComponent>
  </styled.input>
</ThemeProvider>
`;

exports[`FormElements Textfield renders correctly in disabled state 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.input
    disabled={true}
    value="Disabled"
  >
    <StyledComponent
      disabled={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      value="Disabled"
    >
      <input
        className="c0"
        disabled={true}
        value="Disabled"
      />
    </StyledComponent>
  </styled.input>
</ThemeProvider>
`;

exports[`FormElements Textfield renders correctly in error state 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
  border-color: #EF4423;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.input
    hasError={true}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      hasError={true}
    >
      <input
        className="c0"
      />
    </StyledComponent>
  </styled.input>
</ThemeProvider>
`;

exports[`FormElements Textfield renders correctly with a value 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  background-color: #FFFFFF;
  border-radius: 2px;
  border: 1px solid #DBDBDB;
  padding: 0 16px;
  outline: none;
  box-shadow: none;
  -webkit-transition: all .2s ease;
  transition: all .2s ease;
  font-size: 14px;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  color: #111111;
}

.c0:hover {
  border-color: #949494;
}

.c0:focus {
  border-color: #111111;
}

.c0:disabled {
  color: #DBDBDB;
  border-color: #DBDBDB;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "fontFamily": "Open Sans",
      },
      "media": Object {
        "desktop": [Function],
        "laptop": [Function],
        "phone": [Function],
        "tablet": [Function],
      },
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.input
    disabled={true}
    value="I am a textfield"
  >
    <StyledComponent
      disabled={true}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      value="I am a textfield"
    >
      <input
        className="c0"
        disabled={true}
        value="I am a textfield"
      />
    </StyledComponent>
  </styled.input>
</ThemeProvider>
`;
